### ISSUE TITLE
Readonly structs should warn when fields are implicitly copied for member invocation

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#33968]
The following doesn't produce a warning today, but eventually it should in a warning wave. A similar version where `S1.s2` and `S1` are non-readonly but `S1.M1` is readonly using the readonly members feature (see dotnet/csharplang#1710) should produce the same warning.

```cs
public readonly struct S1
{
    public readonly S2 s2;
    public void M1()
    {
        // warn on implicit local copy
        s2.M2();

        // no warning on explicit copy
        var copy = s2;
        copy.M2();
    }
}

public struct S2
{
    public int i;
    public void M2()
    {
        i = 23;
    }
}
```

**Comment by benaadams** [comment]
Related https://github.com/dotnet/roslyn/issues/17310

**Comment by RikkiGibson** [comment]
Note that we've decided to give a warning out of the box when a non-readonly member is called from a readonly member and the receiver is `this`. Implicit copy warnings on fields will only be handled by this warning wave, regardless of whether the copy is occurring due to the field being readonly or the containing method being readonly.

This is in part due to the desire to have uniformity between readonly structs where the methods are also marked readonly, and readonly structs where the methods are not marked readonly.

**Comment by RikkiGibson** [comment]
@jaredpar @jmarolf should we put this in wave 6?

**Comment by jaredpar** [comment]
It should definitely be a candidate

**Comment by jmarolf** [comment]
I have no objections to this being in Wave 6. In general, if the compiler team thinks a warning is ok I am certainly not going to have a difference in opinion.
